Composer/packages/client/src/router.tsx

```diff
@@ -41,79 +43,118 @@ const Publish = React.lazy(() => import('./pages/publish/Publish'));
+`;
+
 const Routes = (props) => {
   const botOpening = useRecoilValue(botOpeningState);
   const pluginPages = useRecoilValue(pluginPagesSelector);
   const spinnerText = useRecoilValue(botOpeningMessage);
+  const showDebugPanel = useDebugPane();
 
   return (
-    <div css={data}>
-      <Suspense fallback={<LoadingSpinner />}>
-        <Router basepath={BASEPATH} {...props}>
-          <Redirect
-            noThrow
-            from="/bot/:projectId/language-generation"
-            to="/bot/:projectId/language-generation/common"
-          />
-          <Redirect
-            noThrow
-            from="/bot/:projectId/language-understanding"
-            to="/bot/:projectId/language-understanding/all"
-          />
-          <Redirect noThrow from="/bot/:projectId/knowledge-base" to="/bot/:projectId/knowledge-base/all" />
-          <Redirect noThrow from="/bot/:projectId/publish" to="/bot/:projectId/publish/all" />
-          <Redirect noThrow from="/" to={resolveToBasePath(BASEPATH, 'home')} />
-          <ProjectRouter path="/bot/:projectId/skill/:skillId">
-            <DesignPage path="dialogs/:dialogId/*" />
-            <LUPage path="language-understanding/:dialogId/item/:luFileId/*" />
-            <LUPage path="language-understanding/:dialogId/*" />
-            <LGPage path="language-generation/all/*" />
-            <LGPage path="language-generation/:dialogId/item/:lgFileId/*" />
-            <LGPage path="language-generation/:dialogId/*" />
-            <QnAPage path="knowledge-base/:dialogId/item/:qnaFileId/*" />
-            <QnAPage path="knowledge-base/:dialogId/*" />
-            <BotProjectSettings path="botProjectsSettings" />
-            <Diagnostics path="diagnostics" />
-            <DesignPage path="*" />
-          </ProjectRouter>
-          <ProjectRouter path="/bot/:projectId">
-            <DesignPage path="dialogs/:dialogId/*" />
-            <LUPage path="language-understanding/:dialogId/item/:luFileId/*" />
-            <LUPage path="language-understanding/:dialogId/*" />
-            <LGPage path="language-generation/all/*" />
-            <LGPage path="language-generation/:dialogId/item/:lgFileId/*" />
-            <LGPage path="language-generation/:dialogId/*" />
-            <QnAPage path="knowledge-base/:dialogId/*" />
-            <Publish path="publish/:targetName" />
-            <BotProjectSettings path="botProjectsSettings" />
-            <FormDialogPage path="forms/:schemaId/*" />
-            <FormDialogPage path="forms/*" />
-            <DesignPage path="*" />
-            <Diagnostics path="diagnostics" />
-            {pluginPages.map((page) => (
-              <PluginPageContainer
-                key={`${page.id}/${page.bundleId}`}
-                bundleId={page.bundleId}
-                path={`plugin/${page.id}/${page.bundleId}`}
-                pluginId={page.id}
+    <div css={root}>
+      <div css={pageWrapper}>
+        <div css={contentWrapper}>
+          <WebChatContainer />
```

Comment:
can it be lazy loaded (at least heavy part of it)? it's only visible on demand

Reply:
so the actual component is lazy loaded actually. It is shown only when the bot is started and a conversation ID is set. Having this here would basically just spin up a new directline server if none is available. No UI elements to delay the loading of the app until bot is started.